torch/_inductor/inductor_prims.py

```diff
@@ -71,3 +72,24 @@ def eager_force_stride(input_tensor, stride):
     lambda input_tensor, stride: eager_force_stride(input_tensor, stride),
     doc="Force the stride order for input tensor. No-op if the input tensor already has the stride. Do a copy otherwise",
 )
+
+
+def _inductor_bucketize_impl(input, boundaries, *, out_int32=False, right=False):
+    return torch.bucketize(input, boundaries, out_int32=out_int32, right=right)
+
+
+def _inductor_bucketize_meta(input, boundaries, *, out_int32=False, right=False):
+    return torch.empty_like(
+        input,
+        memory_format=torch.preserve_format,
+        dtype=(torch.int32 if out_int32 else torch.int64),
+    )
+
+
+_bucketize = _prims._make_prim(
+    schema="_inductor_bucketize(Tensor input, Tensor boundaries, *, bool out_int32=False, bool right=False) -> Tensor",
```

Comment:
What's the advantage of adding a prim rather than just not decomposing `aten.bucketize` in inductor?

Reply:
because this implementation is not currently efficient, I'm the lowering for prims._inductor_bucketize so that it doesn't affect perf for models that use torch.bucketize. After we make the implementation more efficient, we'll remove prims._inductor_bucketize and add the lowering directly to torch.bucketize.

Reply:
Oh sorry for the noise, should have read the PR description more carefully.